onnxruntime/core/graph/model.cc

```diff
@@ -377,6 +381,15 @@ ModelProto Model::ToProto() const {
   // out dense duplicates of sparse initializers and leave the original
   // proto intact.
   ModelProto result(model_proto_);
+
+  // Sync current model_metadata_ back to protobuf metadata_props
+  result.clear_metadata_props();
```

Comment:
Perhaps we also need to do this in ` Model::ToGraphProtoWithExternalInitializers`